File: DesignPatternsExercises/java/FacadeDesignPattern/Exercise/FacadeDesignPattern.java
public class FacadeDesignPattern {
	public static void main(String[] args) {
		System.out.println("Student ID: 2001 and Book: " + Constants.JAVA_BOOK);
		BookCheckoutFacade bookCheckoutFacade = new BookCheckoutFacade(2001);
		bookCheckoutFacade.issueBook(Constants.JAVA_BOOK);
		System.out.println();

		System.out.println("Student ID: 2002 and Book: " + Constants.JAVA_BOOK);
		bookCheckoutFacade = new BookCheckoutFacade(2002);
		bookCheckoutFacade.issueBook(Constants.JAVA_BOOK);
		System.out.println();

		System.out.println("Student ID: 1 and Book: " + Constants.DESIGN_BOOK);
		bookCheckoutFacade = new BookCheckoutFacade(1);
		bookCheckoutFacade.issueBook(Constants.DESIGN_BOOK);
	}
}

File: DesignPatternsExercises/ruby/facade_design_pattern/solution/facade_design_pattern.rb
require_relative 'book_checkout_facade.rb'

module Constants
  JAVA_BOOK = 'Java - The Complete Reference'.freeze
  DESIGN_BOOK = 'Design of Wood Structures'.freeze
  MISTAKES_BOOK = 'Three Mistakes of My Life'.freeze
end

class FacadeDesignPattern
  # BookAvailability.__________(1)__________
  BookAvailability.create_book_map

  puts "Student ID: 2001 and Book: #{Constants::JAVA_BOOK}"
  book_checkout_facade = BookCheckoutFacade.new(2001)
  book_checkout_facade.issue_book(Constants::JAVA_BOOK)
  puts ''

  puts "Student ID: 2002 and Book: #{Constants::JAVA_BOOK}"
  book_checkout_facade = BookCheckoutFacade.new(2002)
  book_checkout_facade.issue_book(Constants::JAVA_BOOK)
  puts ''

  puts "Student ID: 1 and Book: #{Constants::DESIGN_BOOK}"
  book_checkout_facade = BookCheckoutFacade.new(1)
  book_checkout_facade.issue_book(Constants::DESIGN_BOOK)
end
